### ISSUE TITLE
[BITV]: remove all native tooltips from breadcrumbs

### ISSUE THREAD

**Issue by JuliaKirschenheuter** [nextcloud/server#42460]
I think it would be better to remove native tooltips from breadcrumbs (all `title` attrs) because then we will face with a problem again: tooltips are there but there is no other way to discover this information besides mouse over.

![Screenshot from 2023-12-22 16-53-30](https://github.com/nextcloud/server/assets/6078378/7b713ab9-9dde-4a85-8593-7b085b24f368)

**Comment by JuliaKirschenheuter** [comment]
This issue https://github.com/nextcloud/server/issues/42265 was there because if we provide an info via `title` then we have provide the same info via `aria-label`

**Comment by ShGKme** [comment]
> then we will face with a problem again: tooltips are there but there is no other way to discover this information besides mouse over

Is it a problem?

As [commented in the PR](https://github.com/nextcloud/server/pull/42379#pullrequestreview-1790054847), I agree that most of these `title`s here unneeded because there is enough information already. Except probably the first one (with Home icon) and the last one (because it only reloads content).

But it should not be an issue, because `title`-s are not the only interface to discover this information.

**Comment by JuliaKirschenheuter** [comment]
> Is it a problem?

If the info is only available via title - we have a11y problem. I mean in that case it is more "decorative than informative" tooltip, but to be sure lets remove it. 

First one and lats one could still have a title as far they also have `aria-label` too.

**Comment by skjnldsv** [comment]
I'd like o challenge this, I think the title should stay.

Following an important message from @JuliaKirschenheuter and @ShGKme :
>>> then we will face with a problem again: tooltips are there but there is no other way to discover this information besides mouse over
>>>
>> Is it a problem?
>
> If the info is only available via title - we have a11y problem. I mean in that case it is more "decorative than informative" tooltip, but to be sure lets remove it.
> First one and last one could still have a title as far they also have aria-label too.

Removing that mouse tooltip is reducing visibility for users.
Aria ressources are not available for people that don't use accessibility tools. I'd rather have a tooltip AND aria-label than only one or none of the two.

In a more global way, screen readers behaviour is far less fragmented than it used to be.
Most of them are reading title and aria-label (or ignore one of the two if they have the same string)
here is how Firefox (and checked with Chrome too) sees the entire NAV
![image](https://github.com/nextcloud/server/assets/14975046/2710971e-4af3-4f8d-8709-177fe9934279)

You can see that the links are properly handled and aria-label should not even be necessary (while as good measure, I think BOTH should be used.

**Comment by ShGKme** [comment]
> Removing that mouse tooltip is reducing visibility for users.

There are 3 types of breadcrumbs here:
1. 🏚️ = `Home` in the beginning
2. "Folder name" with `Go to "/path/to/Folder Name"` tooltip
3. The last button with "Reload current directory"

I agree with 1 and maybe 3 (as it has kind of a custom handling), but in 2 it doesn't show any necessary information imo.

> Most of them are reading title and aria-label (or ignore one of the two if they have the same string)

Yes, and that is also a problem when we have unnecessary titles. With a screen reader, you are overwhelmed with information for each link. In the worth case, it could be `Clickable visited link Baz go to the slash lorem slash ipsum slash foo slash bar slash baz directory`.

But it should be fine for most modern screen readers that ignore `title` when there is a text content.

> You can see that the links are properly handled and aria-label should not even be necessary (while as good measure, I think BOTH should be used.

We have `aria-label` only on the first button `🏚️ Home` where there is no text content. It is necessary there, because `title` doesn't replace `aria-label` (is not a label), and it doesn't work well in some screen-readers. Some ignore them, some read them even when they duplicate `aria-label`. Command `Click on Go to the "/" directory"` won't work in voice assistant.

So `title` does not replace `aria-label`.

**Comment by JuliaKirschenheuter** [comment]
Hi @skjnldsv 

the initial issue we had: https://github.com/nextcloud/server/issues/42265. That means that all `aria-labels` on a navigation item with a text have to disappear. Second step is: if there is no `aria-label` which has served as important info for screen reader users then all `title`'s have to be removed too (we can't leave items which are discoverable only via `title`).

That means we can leave `title` + `aria-label` only for icon-only navigation items.

Thank you @ShGKme for detailed explanation!

@skjnldsv is there something in your eyes how can we reach this point regarding implementation?

**Comment by ShGKme** [comment]
> (we can't leave items which are discoverable only via `title`).

We don't have such buttons.

Only `🏚️` (`Home`) button doesn't have a text and is not discoverable by text content, but it has an `aria-label`.

Other buttons have implicit text.
